lib/realtime/block_waiver.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Realtime.BlockWaiver do
+  @moduledoc """
+  A representation of known skipped work on a trip, entered by dispatch.
+  Essentially this summarizes a set of StopTimeUpdate datapoints.
+  """
+
+  alias Concentrate.StopTimeUpdate
+  alias Gtfs.{Block, Trip}
+  alias Realtime.Server
+
+  @type t :: %__MODULE__{
+          trip_id: Trip.id(),
+          start_time: Util.Time.time_of_day(),
+          end_time: Util.Time.time_of_day(),
+          remark: String.t() | nil
+        }
+
+  @type block_waivers_by_trip :: %{Trip.id() => t()}
+
+  @enforce_keys [:trip_id, :start_time, :end_time]
+
+  @derive Jason.Encoder
+
+  defstruct [:trip_id, :start_time, :end_time, :remark]
```

Comment:
Ok. Is there a way to specify that in the formatter? I'm a big believer in the text editor being in charge of formatting, not me.

Reply:
Looks like no. [docs](https://hexdocs.pm/mix/master/Mix.Tasks.Format.html) and [docs](https://hexdocs.pm/elixir/Code.html#format_string!/2). Especially this:

> The formatter does not hard code names. The formatter will not behave specially because a function is named defmodule, def, or the like.

So it wouldn't be able to treat the list after `defstruct` any different from any other list.